### DISCUSSION TITLE
Questions about Dispatcher methods

### DISCUSSION THREAD

**Discussion by laolarou726** [AvaloniaUI/Avalonia#15359] [General]
Can someone explain to me the differences between those Dispatcher methods?:

- `Dispatcher.UIThread.Post()`
- `Dispatcher.UIThread.Invoke()`
- `Dispatcher.UIThread.InvokeAsync()`

Especially the differences between `Dispatcher.UIThread.Post()` and `Dispatcher.UIThread.Invoke()`. These two functions confused me. I'm currently mixed using those two methods in my project.

**Comment by timunie** [comment]
Use Post if you don't need the return value, which is most of the time the case imo.
